packages/ui-utils/src/process-chat-response.ts

```diff
@@ -71,7 +71,7 @@ export async function processChatResponse({
     // trigger update for streaming by copying adding a revision id that changes
     // (without it, the changes get stuck in SWR and are not forwarded to rendering):
     const copiedMessage = {
-      ...currentMessage,
+      ...structuredClone(currentMessage), // deep copy
```

Comment:
@ian-pascoe is this key to your fix or optional?

Reply:
This is actually the most important part of the fix

Reply:
wait a sec - we are already shallow cloning, wouldn't that help with referential integrity? this is why `revisionId` exists (for SWR).

Reply:
If you remove the deep clone and use the solidjs tool example or the solidjs annotations example, you'll see that those values do not get updates